### PR TITLE
Media: Replace getNextPageQuery flux selector with redux

### LINKED ISSUE #44143: Media: Replace flux getNextPageQuery with a redux selector
This selector can be easily re-written in redux by doing something along the lines of the following:

1. Create a new action to set the next page handle
1. Dispatch the action in `requestMediaSuccess` with the `data.meta`
1. Create a selector to retrieve the next page handle
1. Create a second selector to use the next page handle to create a new query dervied from the existing query stored in the MediaQueryManager found at `state.media.queries[ siteId ]`

### PR DESCRIPTION
**Pull request by sarayourfriend**
#### Changes proposed in this Pull Request

* Add new redux selectors for next page query
* Track the next page handle in the `fetching` selector (is there a better place for this?)
* Use the new redux selectors for the next page query rather than flux
* Remove the flux stuff

#### Testing instructions

* Using a site with very many media items (more than 50) ensure infinite scroll still works and that the initial page load is 20 items
* Do the same for pexels and google photos
* Also because I made changes to the `getMediaItem` selector, ensure that adding media items to posts in the classic editor still works and that the images appear when you reload the page after saving the draft or publishing (this should be sufficient to test those changes).

A note about Google Photos: the service is sometimes not working correctly (which you can see in the response from the API) so it's worth comparing with prod.

Part of https://github.com/Automattic/wp-calypso/issues/43663

### CODE REVIEW

**Comment by saramarcondes** [comment]
Lint errors are expected and pre-existing. I've cleaned them up in a separate PR so I want to avoid the merge conflict if I also clean them up here.

**Comment by flootr** [comment]
Loading Google Photos library and Pexels doesn't work for me on this branch. Switching e.g. to Pexels results in a request to `/media` but it should be `/meta/external-media/pexels` (which depends on the `query.source` attribute).

This is what the `query` object passed to the data layer handler looks like for me on master:
Screenshot 2020-07-21 at 13 58 11

On this branch it only contains the `page_handle` attribute.

File: client/state/selectors/get-next-page-query.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @param {object} state The state object
+ * @param {number} siteId The site ID
+ */
+export default function getNextPageQuery( state, siteId ) {
+	const queryManager = getMediaQueryManager( state, siteId );
+	if ( ! queryManager ) {
+		return DEFAULT_QUERY;
+	}
+
+	const currentQuery = queryManager.options.query;
```
Comment (flootr) [inline client/state/selectors/get-next-page-query.js:21]: [Regarding my comment here](https://github.com/Automattic/wp-calypso/pull/44149#issuecomment-661812840) it seems like relevant query information is stored in the `queryRequests` reducer

Screenshot 2020-07-21 at 14 10 24

The `queryManager.options.query` returned `undefined` for me both on the initial source change and when filling in the a search field.

Comment (saramarcondes) [reply client/state/selectors/get-next-page-query.js:21]: I think `queryRequests` is the "current" request, not the full query object.

I'm really confused why `queryManager.options.query` isn't defined or why this isn't working as it was working when I first wrote it, but I wonder if I ran into an error and was blowing through compilation errors without noticing 😞

Based on https://github.com/Automattic/wp-calypso/blob/4ecbc4965a9cca5b90a09954d9a7276ea2d16726/client/state/media/reducer.js#L171 I think that `query` should be in the QueryManager, but maybe there's something wrong with this action 🤔

Comment (flootr) [reply client/state/selectors/get-next-page-query.js]: should be fixed with 4b0682e

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-source".

@flootr please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/828476). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: client/state/selectors/test/get-next-page-query.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+				},
+			};
+
+			expect( getNextPageQuery( state, siteId ) ).toEqual( {
+				...query,
+				page_handle: nextPageHandle,
+			} );
+		}
+	);
+} );
```
Comment (saramarcondes) [inline client/state/selectors/test/get-next-page-query.js:51]: These test descriptions need to be updated. I think we can also remove the query manager stuff completely right?

**Comment by scinos** [comment]
* Loaded site with more than 50 items. Infinite scroll is loading, I see actions 'MEDIA_RECEIVED' with batches of 20 items.
* Same thing works with Pexels
* Can't get it to work with Google Photos in live branch, I can see requests to Google returning errors. Something related with it not being served from wordpress.com?
* Adding photos from Media and Pexels to a new post in the classic editor works, including saving, refreshing and previewing.


Probably unrelated:

I Found a bug, probably existing: If you go to Media > Pexels and search for 'thing' and refresh the page, it will try to search for 'thing' in my media, not in Pexels. If I click back to Pexels, the search 'thing' is lost.

**Review by scinos — APPROVED** [review]

**Comment by flootr** [comment]
@scinos I created an issue for that pexels / my media reload bug here https://github.com/Automattic/wp-calypso/issues/44526

**Pull request merged**